src/transformers/modeling_utils.py

```diff
@@ -1569,6 +1587,65 @@ def retrieve_modules_from_names(self, names, add_prefix=False, remove_prefix=Fal
 
         return retrieved_modules
 
+    @classmethod
+    def _low_cpu_mem_usage_load(cls, model, loaded_state_dict_keys, resolved_archive_file):
+
+        # dematerialize param storage for keys that are going to be replaced by state_dict, by
+        # putting those on the meta device
+        for k in loaded_state_dict_keys:
+            split_name = k.split(".")
+            m = model
+            matched = False
+            while len(split_name) > 1:
```

Comment:
nit: I would  extract that in a special function since you do it in another function and it might not be obvious what it is doing.

Reply:
Agreed! 

I was thinking to fold the whole structure and pass some kind of callback, but it'd make things unnecessarily complex for the reader to understand.  so I just refactored the part in your suggestion.